test/unit/math/mix/mat/fun/columns_dot_product_test.cpp

```diff
@@ -32,14 +32,15 @@ TEST(AgradMixMatrixColumnsDotProduct, vector_vector_fv_1stDeriv) {
 
   AVEC z = createAVEC(a.val(), b.val(), c.val(), d.val(), e.val(), f.val());
   VEC h;
-  columns_dot_product(vv_1, vd_2)(0).val_.grad(z, h);
+  columns_dot_product(vv_1, vd_2)(0).val().grad(z, h);
   EXPECT_FLOAT_EQ(4.0, h[0]);
   EXPECT_FLOAT_EQ(-2.0, h[1]);
   EXPECT_FLOAT_EQ(-1.0, h[2]);
   EXPECT_FLOAT_EQ(0.0, h[3]);
   EXPECT_FLOAT_EQ(0.0, h[4]);
   EXPECT_FLOAT_EQ(0.0, h[5]);
 }
+/*
```

Comment:
I need to go through and double check, but a lot of these tests failed at run time where they now fail at compile time (since a lot of these methods now check in the their template that the Eigen matrices are vectors